src/link.ts

```diff
@@ -0,0 +1,234 @@
+import {
+  GraphQLRequest,
+  NextLink,
+  Operation,
+  RequestHandler,
+  FetchResult,
+} from './types';
+
+import {
+  validateOperation,
+} from './linkUtils';
+
+import {
+  parse,
+} from 'graphql/language/parser';
+
+import * as Observable from 'zen-observable';
+import { DocumentNode } from 'graphql/language/ast';
+
+export abstract class ApolloLink {
+
+  public static from(links: (ApolloLink | RequestHandler)[]) {
+    if (links.length === 0) {
+      return ApolloLink.empty();
+    }
+
+    return links
```

Comment:
@jbaxleyiii The issue I ran into was that reduce is applied, so that the initial argument is the first in the chain, so the function calls became:

```js
ApolloLink.empty().concat(first).concat(second)... 
```

We could do an ApolloLink.passthrough() as the initial value